subsys/net/lib/nrf_cloud/src/nrf_cloud_fota_common.c

```diff
@@ -479,8 +516,67 @@ bool nrf_cloud_fota_is_type_enabled(const enum nrf_cloud_fota_type type)
 		return IS_ENABLED(CONFIG_NRF_CLOUD_FOTA_TYPE_MODEM_DELTA_SUPPORTED);
 	case NRF_CLOUD_FOTA_MODEM_FULL:
 		return IS_ENABLED(CONFIG_NRF_CLOUD_FOTA_TYPE_MODEM_FULL_SUPPORTED);
+	case NRF_CLOUD_FOTA_SMP:
+		return IS_ENABLED(CONFIG_NRF_CLOUD_FOTA_TYPE_SMP_SUPPORTED);
 	default:
 		LOG_WRN("Unhandled FOTA type: %d", type);
 		return false;
 	}
 }
+
+int nrf_cloud_fota_smp_version_read(void)
+{
+#if defined(CONFIG_NRF_CLOUD_FOTA_SMP)
+	struct mcumgr_image_state image_list = {0};
+	struct mcumgr_image_data *list;
+	size_t ver_len;
+	int err = dfu_target_smp_image_list_get(&image_list);
+
+	if (err) {
+		LOG_WRN("Failed to read SMP image list, error: %d", err);
+		return -ENODATA;
+	}
+
+	list = image_list.image_list;
+	for (int i = 0; i < image_list.image_list_length; ++i) {
+		LOG_DBG("%s Image(%d) slot(%d)",
+			list->flags.active ? "Primary" : "Secondary",
+			list->img_num,
+			list->slot_num);
+		LOG_DBG("  Version: %s", list->version);
+		LOG_DBG("  Bootable(%d) Pending(%d) Confirmed(%d)",
+			list->flags.bootable, list->flags.pending, list->flags.confirmed);
+
+		if (!list->flags.active) {
+			++list;
+			continue;
```

Comment:
maybe it would be nice to list them all, not just stop when you get to an active one?